### PR TITLE
SNV somatic downstream pipeline

### PR DESCRIPTION
**Pull request by dmaziec**

### CODE REVIEW

File: descriptions/driver_annot.cwl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+outputs:
+
+  - id: putative_drivers_vcf
+    type: File
+    outputBinding:
+      glob: $(inputs.output).gz
+    secondaryFiles:
+      - .tbi
+doc: |
+  vcf containing reported putative drivers
```
Comment (B3rse) [inline descriptions/driver_annot.cwl]:
```suggestion
  run driverCatalogVCF to annotate putative driver genes in input VCF file
```

File: descriptions/dump_drivers.cwl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+outputs:
+
+  - id: putative_drivers_json
+    type: File
+    outputBinding:
+      glob: $(inputs.output)
+    secondaryFiles:
+      - .tbi
+doc: |
```
Comment (B3rse) [inline descriptions/dump_drivers.cwl:45]:
```suggestion

doc: |
```

File: descriptions/driver_annot.cwl [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+outputs:
+
+  - id: putative_drivers_vcf
+    type: File
+    outputBinding:
+      glob: $(inputs.output).gz
+    secondaryFiles:
+      - .tbi
+doc: |
```
Comment (B3rse) [inline descriptions/driver_annot.cwl]:
```suggestion

doc: |
```

File: descriptions/dump_drivers.cwl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  JSON including reported putative drivers
```
Comment (B3rse) [inline descriptions/dump_drivers.cwl]:
```suggestion
  run dumpJSON to create a JSON file with the reported putative driver genes
```

File: descriptions/somatic_sentieon_vcf_splitter.cwl [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    doc: expect the path to the somatic vcf gz file
+
+  - id: prefix
+    type: string
+    inputBinding:
+      prefix: -p
+    default: 'pre'
+    doc: base name of the full PASS-filtered output vcf gz file
+
+
```
Comment (B3rse) [inline descriptions/somatic_sentieon_vcf_splitter.cwl]:
```suggestion
```

File: descriptions/somatic_sentieon_vcf_splitter.cwl [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+outputs:
+
+  - id: indel_snv_pass
+    type: File
+    outputBinding:
+      glob: $(inputs.prefix + "_ind_snv.vcf.gz")
+    secondaryFiles:
+      - .tbi
+  - id: sv_bnd_pass
```
Comment (B3rse) [inline descriptions/somatic_sentieon_vcf_splitter.cwl]:
```suggestion

  - id: sv_bnd_pass
```

File: descriptions/somatic_sentieon_vcf_splitter.cwl [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    type: File
+    outputBinding: 
+      glob: $(inputs.prefix + "_bnd.vcf.gz")
+    secondaryFiles:
+      - .tbi
+doc: |
```
Comment (B3rse) [inline descriptions/somatic_sentieon_vcf_splitter.cwl]:
```suggestion

doc: |
```

File: dockerfiles/driver_catalog/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,45 @@
+#######################################################################
+#     Basic image
+#######################################################################
+FROM cgap/cgap-ubuntu2004-py-38:0.0.1
+MAINTAINER Michele Berselli (berselli.michele@gmail.com), Dominika Maziec (dominika_maziec@hms.harvard.edu)
+
+#######################################################################
+#     General updates & installing necessary Linux components
+#       - autoconf, pkg-config used by vcftools
```
Comment (B3rse) [inline dockerfiles/driver_catalog/Dockerfile]:
```suggestion
#       - autoconf, pkg-config
#.      - bcftools
```

File: dockerfiles/driver_catalog/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+#######################################################################
+#     Software
+#######################################################################
+## granite pandas
+RUN pip install granite-suite==0.2.0 pandas==1.5.2
+
+#######################################################################
+#     Scripts
+#######################################################################
+
```
Comment (B3rse) [inline dockerfiles/driver_catalog/Dockerfile:33]:
```suggestion
```

File: dockerfiles/driver_catalog/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (B3rse) [inline dockerfiles/driver_catalog/Dockerfile]:
```suggestion
```

File: dockerfiles/driver_catalog/scripts/somatic_annot.py [hunk 11]
```diff
@@ -0,0 +1,577 @@
...
+        tumor_sample = vnt_obj.IDs_genotypes[0]
+        self.allele_fraction = vnt_obj.get_genotype_value(tumor_sample, "AF")
+
+        super().__init__(category="mutational", **kwargs)
+
+    def to_dict(self):
+        """
+
+        A function used to serialize the object into a dictionary.
+        It is used to dump the varian into a JSON format, some of the parameters are not included.
```
Comment (B3rse) [inline dockerfiles/driver_catalog/scripts/somatic_annot.py]:
```suggestion
        It is used to dump the variant into a JSON format, some of the parameters are not included.
```

File: dockerfiles/driver_catalog/scripts/somatic_annot.py [hunk 12]
```diff
@@ -0,0 +1,577 @@
...
+        self.chrom = chrom
+        self.start = start
+        self.end = end
+        super().__init__(**kwargs)
+
+    def to_dict(self):
+        """
+
+        A function used to serialize the object into a dictionary.
+        It is used to dump the varian into the JSON format.
```
Comment (B3rse) [inline dockerfiles/driver_catalog/scripts/somatic_annot.py]:
```suggestion
        It is used to dump the variant into the JSON format.
```

File: dockerfiles/snv_somatic_splitter/scripts/somatic_sentieon_vcf_splitter.py [hunk 13]
```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+################################################
+#
+#  Script to split Sentieon somatic vcf 3 ways
```
Comment (B3rse) [inline dockerfiles/snv_somatic_splitter/scripts/somatic_sentieon_vcf_splitter.py]:
```suggestion
#  Script to split Sentieon somatic vcf by variant type
```

File: dockerfiles/snv_somatic_splitter/scripts/somatic_sentieon_vcf_splitter.py [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+#   and filter for high confidence variants
```
Comment (B3rse) [inline dockerfiles/snv_somatic_splitter/scripts/somatic_sentieon_vcf_splitter.py]:
```suggestion
```

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+        argument_type: file.vcf_gz
+
+      CADD_snv:
+        argument_type: file.tsv_gz
+
+      CADD_indel:
+        argument_type: file.tsv_gz
+
+      phylop30bw:
+        argument_type: file.BigWig
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]:
```suggestion
        argument_type: file.bigWig
```

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 16]
```diff
@@ -0,0 +1,275 @@
...
+
+      phylop100bw:
+        argument_type: file.BigWig
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]:
```suggestion
        argument_type: file.bigWig
```

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 17]
```diff
@@ -0,0 +1,275 @@
...
+
+      phastc100bw:
+        argument_type: file.BigWig
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]:
```suggestion
        argument_type: file.bigWig
```

**Review by B3rse — CHANGES_REQUESTED** [review]

**Review by dominikglodzikhms — COMMENTED** [review]
Comments now look comprehensive - thank you for making these changes!

File: descriptions/dump_drivers.cwl [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    dockerPull: ACCOUNT/driver_catalog:VERSION
+
+baseCommand: [python3, /usr/local/bin/somatic_annot.py, dumpJSON ] 
+
+inputs:
+  - id: vcf
+    type: File
+    inputBinding:
+      prefix: -v
+    doc: expect the path to the somatic vcf gz file
```
Comment (B3rse) [inline descriptions/dump_drivers.cwl]: is this the driver file?

File: descriptions/dump_drivers.cwl [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+
+  - id: cnv
+    type: File
+    inputBinding:
+      prefix: -c
+    doc: SV file containing somatic CNVs
```
Comment (B3rse) [inline descriptions/dump_drivers.cwl]: probably we can specify that these are the putative drivers

File: portal_objects/metaworkflows/SNV_CNV_dump.yaml [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+
+      cnv:
+        argument_type: file.tsv_gz
+        input_dimension: 1
+    
+    output:
+
+      putative_drivers_json:
+        description: GosCan compatible file containing driver mutations for SNVs, INDELs and CNVs
+        file_type: JSON
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_CNV_dump.yaml]: I would do more something like `driver JSON` so we have a bit of information of what is in the file

File: portal_objects/metaworkflows/SNV_CNV_dump.yaml [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+    
+    config:
+        instance_type:
+        - t3.large
+        - t3.xlarge
+        ebs_size: 1.6x
+        EBS_optimized: True
+        spot_instance: False
+        run_name: run_drivers_dump
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_CNV_dump.yaml]:
```suggestion
        run_name: run_dump_drivers
```

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 22]
```diff
@@ -0,0 +1,275 @@
+## Pipeline information #####################################
+#     General information for the pipeline
+#############################################################
+name: SNV_somatic_downstream
+description: Downstream pipeline to annotate somatic variants
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]:
```suggestion
description: Downstream pipeline to annotate somatic variants and flag putative driver mutations.
```

File: portal_objects/metaworkflows/SNV_CNV_dump.yaml [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+  vcf:
+    argument_type: file.vcf_gz
+    dimensionality: 1
+  
+  cnv:
+    argument_type: file.tsv_gz
+    dimensionality: 1 
+
+
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_CNV_dump.yaml]:
```suggestion
```

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 24]
```diff
@@ -0,0 +1,275 @@
+## Pipeline information #####################################
+#     General information for the pipeline
+#############################################################
+name: SNV_somatic_downstream
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]: can we rename the file to be consistent with the name of the pipeline?

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 25]
```diff
@@ -0,0 +1,275 @@
...
+    files:
+      - cadd-snv@1.6
+
+  CADD_indel:
+    argument_type: file.tsv_gz
+    files:
+      - cadd-indel@1.6
+
+  phylop30bw:
+    argument_type: file.BigWig
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]: is this bigWig?

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 26]
```diff
@@ -0,0 +1,275 @@
...
+    files:
+      - phylop30way-mammalian@hg38
+  
+  phylop100bw:
+    argument_type: file.BigWig
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]: same bigWig

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 27]
```diff
@@ -0,0 +1,275 @@
...
+    files:
+      - phylop100way-vertebrate@hg38
+
+  phastc100bw:
+    argument_type: file.BigWig
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]: same ^

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 28]
```diff
@@ -0,0 +1,275 @@
...
+
+  version: 
+    argument_type: parameter.string
+    value: "101"
+
+  assembly:
+    argument_type: parameter.string
+    value: "GRCh38"
+
+
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]:
```suggestion
```

File: portal_objects/metaworkflows/SNV_CNV_dump.yaml [hunk 29]
```diff
@@ -0,0 +1,56 @@
+## Pipeline information #####################################
+#     General information for the pipeline
+#############################################################
+name: drivers_to_JSON
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_CNV_dump.yaml]: can we name the script as the pipeline name?

File: portal_objects/metaworkflows/SNV_CNV_dump.yaml [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+        description: GosCan compatible file containing driver mutations for SNVs, INDELs and CNVs
+        file_type: JSON
+    
+    config:
+        instance_type:
+        - t3.large
+        - t3.xlarge
+        ebs_size: 1.6x
+        EBS_optimized: True
+        spot_instance: False
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_CNV_dump.yaml]: this probably should the True once it's tested

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 31]
```diff
@@ -0,0 +1,275 @@
...
+    ## EC2 Configuration to use ########
+    ####################################
+    config:
+        instance_type: 
+         - t3.large
+         - t3.xlarge
+         - c5.4xlarge
+        ebs_size: 1.6x
+        EBS_optimized: True
+        spot_instance: False
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]: this probably should the True once it's tested

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 32]
```diff
@@ -0,0 +1,275 @@
...
+
+    ## EC2 Configuration to use ########
+    ####################################
+    config:
+        instance_type:
+          - m6a.24xlarge
+          - m6a.32xlarge
+        ebs_size: 1.35x
+        EBS_optimized: True
+        spot_instance: False
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]: this probably should the True once it's tested

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 33]
```diff
@@ -0,0 +1,275 @@
...
+        run_name: run_vep
+        behavior_on_capacity_limit: wait_and_retry
+  
+
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]:
```suggestion
```

File: portal_objects/metaworkflows/SNV_annot.yaml [hunk 34]
```diff
@@ -0,0 +1,275 @@
...
+    ## EC2 Configuration to use ########
+    ####################################
+    config:
+        instance_type:
+        - t3.large
+        - t3.xlarge
+        ebs_size: 1.6x
+        EBS_optimized: True
+        spot_instance: False
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_annot.yaml]: this probably should the True once it's tested

File: portal_objects/workflows/splitter.yaml [hunk 35]
```diff
@@ -0,0 +1,35 @@
+## Workflow information #####################################
+#     General information for the workflow
+#############################################################
+name: splitter
+description: Separete SNVs and INDELs from SVs.
```
Comment (B3rse) [inline portal_objects/workflows/splitter.yaml]: Separate

**Review by B3rse — CHANGES_REQUESTED** [review]

File: portal_objects/metaworkflows/SNV_annot.yaml (hunk 31 above)
Comment (dmaziec) [reply portal_objects/metaworkflows/SNV_annot.yaml]: will switch it off everywhere after the final testing

**Review by B3rse — APPROVED** [review]

File: portal_objects/metaworkflows/drivers_to_JSON.yaml [hunk 36]
```diff
@@ -0,0 +1,61 @@
...
+        file_type: driver JSON
+    
+    config:
+        instance_type:
+        - t3.large
+        - t3.xlarge
+        ebs_size: 1.6x
+        EBS_optimized: True
+        spot_instance: True
+        run_name: run_dump_drivers
```
Comment (B3rse) [inline portal_objects/metaworkflows/drivers_to_JSON.yaml:60]:
```suggestion
        run_name: drivers_to_JSON
```

File: portal_objects/metaworkflows/SNV_somatic_downstream.yaml [hunk 37]
```diff
@@ -0,0 +1,276 @@
...
+
+    ## EC2 Configuration to use ########
+    ####################################
+    config:
+        instance_type:
+          - m6a.24xlarge
+          - m6a.32xlarge
+        ebs_size: 1.35x
+        EBS_optimized: True
+        spot_instance: False
```
Comment (B3rse) [inline portal_objects/metaworkflows/SNV_somatic_downstream.yaml]:
```suggestion
        spot_instance: True
```

**Review by B3rse — APPROVED** [review]

**Pull request merged**